nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -145,25 +117,140 @@ def _set_backend_entity(self, model: TModel) -> None:
+        :param is_last_layer_shared: Indicates whether the last layer which shares the weight
+            should be quantized or not.
+        :return: Information about each weight node that is considered for mixed precision.
+        """
+        if self._mode in [CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM] or self._all_layers:
+            return all_weight_params
+
+        ratio_defining_params = list(
+            filter(
+                lambda wp: wp.node_with_weight.metatype not in self._backend_entity.embedding_metatypes,
+                all_weight_params,
+            )
+        )
+        if not is_last_layer_shared:
+            ratio_defining_params = ratio_defining_params[:-1]
+        return ratio_defining_params
+
+    def _set_weight_compression_config(
+        self,
+        ratio_defining_params: List[WeightCompressionParameters],
+        model: TModel,
+        graph: NNCFGraph,
+        activations: Optional[Dict[str, List[Tensor]]] = None,
+    ) -> None:
+        """
+        Sets the appropriate compression configuration for weights based on some criteria.
+
+        :param ratio_defining_params: Information about weights that are used for calculating ratio between primary and
+            backup precisions.
+        :param model: The model.
+        :param graph: The model graph associated with the model.
+        :param activations: The input activations of the layers considered for compression.
+        """
+        primary_config = WeightCompressionConfig(mode=self._mode, group_size=self._group_size)
+        if self._ratio == 1:
+            for weight_param in ratio_defining_params:
+                weight_param.compression_config = primary_config
+        else:
+            criterion_cls = MIXED_PRECISION_CRITERIA.get(self._sensitivity_metric)
+            criterion = criterion_cls(
+                model, graph, self._backend_entity, ratio_defining_params, primary_config, self._ratio, activations
+            )
+            criterion.assign_mixed_precision()
+
+    @staticmethod
+    def _proportion_str(num_weights_list: List[int], total_num_weights: int, total_num_params: int) -> str:
+        """
+        Generates a string with proportion between target parameters and all model parameters by number of weights.
+
+        :param num_weights_list: List of number of weights of target model parameters.
+        :param total_num_weights: The total number of weights.
+        :param total_num_params: The total number of model parameters.
+        :return: The string with proportion between target parameters and all model parameters by number of weights.
+        """
+        percentage = sum(num_weights_list) / max(total_num_weights, 1) * 100
+        return f"{percentage:.0f}% ({len(num_weights_list)} / {total_num_params})"
+
+    def _get_bitwidth_distribution_str(
+        self, all_params: List[WeightCompressionParameters], internal_params: List[WeightCompressionParameters]
```

Comment:
Please take the latest version of this function from develop:
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L193-L205
`internal_params` was renamed

Reply:
Thanks, done.